services/parser-database/connector.py

```diff
@@ -83,3 +152,56 @@ def store_article(article_dict):
     articles_in_memory[article_dict["id"]] = article_dict
     save_keywords_in_memory(get_keywords(article_dict["content"]), article_dict)
     logging.info('Article ' + article_dict["id"] + ' assigned keywords')
+
+
+def store_article_in_db(article_dict):
+    db.collection(u'articles').document(article_dict["id"]).set(article_dict)
+    save_keywords_in_db(get_keywords(article_dict["content"]), article_dict)
+    logging.info('Article ' + article_dict["id"] + ' assigned keywords')
+
+
+def save_keywords_in_db(keywords, article):
+    """Saves the keywords from an article in memory
+
+    Args:
+        keywords (JSON): contains keywords
+        article (Article): article object
+    """
+    for keyword in keywords:
+        frequency = article["content"].count(keyword)
+
+        doc_ref = db.collection(u'keywords').where('keyword', '==', keyword)
+        doc = doc_ref.get()
+        
+        if len(doc) != 0 and doc[0] is not None:
+            from_db = doc[0].to_dict()
+            print(from_db)
+            from_db["matching_articles"][article["id"]] = frequency
+            #print(from_db)
+            db.collection(u'keywords').document(doc[0].id).set(from_db)
+        else:
+            to_send = {"keyword": keyword, "matching_articles": {article["id"]: frequency}}
+            db.collection(u'keywords').add(to_send)
```

Comment:
It looks like you're trying to mirror the same structure that you had for the in-memory implementation, which is not necessarily the best implementation. For example, `keywords` could be a field of the article document that gets queried directly, which eliminates the need to store `keywords` as a separate collection. See [this guide](https://firebase.google.com/docs/firestore/query-data/queries) on how to perform different queries against Firestore, particularly the example for [array membership](https://firebase.google.com/docs/firestore/query-data/queries#array_membership).